--- a/src/checkIn.js	2020-11-21 13:24:04.000000000 +0000
+++ b/src/checkIn.js	2020-12-04 18:13:39.086386528 +0000
@@ -24,7 +24,6 @@
 const Gtk = imports.gi.Gtk;
 
 const CheckInDialog = imports.checkInDialog;
-const FacebookBackend = imports.facebookBackend;
 const FoursquareBackend = imports.foursquareBackend;
 
 var CheckInManager = GObject.registerClass({
@@ -66,9 +65,6 @@
     }
 
     _initBackends() {
-        let facebookBackend = new FacebookBackend.FacebookBackend();
-        this._backends[facebookBackend.name] = facebookBackend;
-
         let foursquareBackend = new FoursquareBackend.FoursquareBackend();
         this._backends[foursquareBackend.name] = foursquareBackend;
     }
@@ -172,7 +168,6 @@
         this.message = null;
         this.place = null;
         this.privacy = null;
-        this.broadcastFacebook = false;
         this.broadcastTwitter = false;
     }
 };
--- a/src/org.gnome.Maps.src.gresource.xml	2020-11-21 13:24:04.000000000 +0000
+++ b/src/org.gnome.Maps.src.gresource.xml	2020-12-04 18:14:25.339039997 +0000
@@ -11,7 +11,6 @@
     <file>contextMenu.js</file>
     <file>epaf.js</file>
     <file>exportViewDialog.js</file>
-    <file>facebookBackend.js</file>
     <file>favoritesPopover.js</file>
     <file>foursquareBackend.js</file>
     <file>foursquareGoaAuthorizer.js</file>
--- a/src/main.js	2020-11-21 13:24:04.000000000 +0000
+++ b/src/main.js	2020-12-04 18:15:01.098855721 +0000
@@ -29,7 +29,6 @@
               'GeocodeGlib': '1.0',
               'Gdk': '3.0',
               'GdkPixbuf': '2.0',
-              'GFBGraph': '0.2',
               'Gio': '2.0',
               'GLib': '2.0',
               'Goa': '1.0',
