--- a/vala/valacodecontext.vala	2019-07-16 09:53:13.000000000 +0000
+++ b/vala/valacodecontext.vala	2019-07-21 15:45:28.178405830 +0000
@@ -573,8 +573,8 @@
 			defines.add ("GLIB_2_%d".printf (i));
 		}
 
-		target_glib_major = glib_minor;
-		target_glib_minor = glib_major;
+		target_glib_major = glib_major;
+		target_glib_minor = glib_minor;
 	}
 
 	public string? get_vapi_path (string pkg) {
--- a/vala/valacodecontext.c	2019-07-16 10:21:14.000000000 +0000
+++ b/vala/valacodecontext.c	2019-07-21 15:45:40.983736088 +0000
@@ -1091,8 +1091,8 @@
 			}
 		}
 	}
-	self->priv->target_glib_major = glib_minor;
-	self->priv->target_glib_minor = glib_major;
+	self->priv->target_glib_major = glib_major;
+	self->priv->target_glib_minor = glib_minor;
 }
 
 gchar*
