--- gnome-weather-40.0/src/app/world.js~	2021-03-20 23:24:49.000000000 +0000
+++ gnome-weather-40.0/src/app/world.js	2021-03-29 12:38:37.008361570 +0000
@@ -212,7 +212,7 @@
 
         info._updatedId = info.connect('updated', (info) => {
             tempLabel.label = info.get_temp_summary();
-            image.icon_name = info.get_symbolic_icon_name();
+            image.icon_name = info.get_icon_name();
         });
 
         this._syncStackPopover();
