--- a/gthumb/gth-application.c	2020-04-19 18:58:10.000000000 +0000
+++ b/gthumb/gth-application.c	2020-11-13 17:47:42.783967428 +0000
@@ -146,7 +146,7 @@
 {
 	G_APPLICATION_CLASS (gth_application_parent_class)->startup (application);
 
-	g_object_set (gtk_settings_get_default (), "gtk-application-prefer-dark-theme", TRUE, NULL);
+	//g_object_set (gtk_settings_get_default (), "gtk-application-prefer-dark-theme", TRUE, NULL);
 
 	gth_pref_initialize ();
 	migrate_data ();
