--- a/src/client/composer/composer-container.vala	2018-08-29 14:57:20.000000000 +0000
+++ b/src/client/composer/composer-container.vala	2018-11-18 15:20:09.818855225 +0000
@@ -14,8 +14,10 @@
 
     // Workaround to retrieve all Gtk.Actions with conflicting accelerators
     protected const string[] conflicting_actions = {
-        GearyController.ACTION_MARK_AS_UNREAD,
-        GearyController.ACTION_FORWARD_MESSAGE
+        GearyController.ACTION_CONVERSATION_LIST,
+        GearyController.ACTION_FORWARD_MESSAGE,
+        GearyController.ACTION_MARK_AS_READ,
+        GearyController.ACTION_MARK_AS_UNREAD
     };
 
     // We use old_accelerators to keep track of the accelerators we temporarily disabled.
@@ -92,8 +94,17 @@
         // Now add our actions to the window and their accelerators
         foreach (string action in ComposerWidget.action_accelerators.get_keys()) {
             this.top_window.add_action(composer.get_action(action));
-            app.set_accels_for_action("win." + action,
-                                      ComposerWidget.action_accelerators[action].to_array());
+            string[] accels =
+                    ComposerWidget.action_accelerators[action].to_array();
+
+            // it seems to_array() is not creating a null-terminated array
+            // as required by Glib, so we do this manually for now
+            string[] accelsn = new string[accels.length + 1];
+            for(int i=0; i<accels.length; i++)
+                accelsn[i] = accels[i];
+            accelsn[accels.length] = null;
+
+            app.set_accels_for_action("win." + action, accelsn);
         }
     }
 
