--- a/meson.build	2018-06-08 03:22:24.000000000 +0000
+++ b/meson.build	2018-09-16 23:07:53.782155115 +0000
@@ -1,6 +1,6 @@
 project('epiphany', 'c',
   license: 'GPL3+',
-  version: '3.30.5',
+  version: '3.30.5.2',
   meson_version: '>= 0.42.0',
   default_options: ['c_std=gnu11']
 )
@@ -42,7 +42,7 @@
 )
 
 glib_requirement = '>= 2.52.0'
-gtk_requirement = '>= 3.22.13'
+gtk_requirement = '>= 3.22.12'
 nettle_requirement = '>= 3.2'
 webkitgtk_requirement = '>= 2.17.4'
 
