--- a/embed/ephy-embed-shell.c	2020-06-25 19:38:44.301610270 +0000
+++ b/embed/ephy-embed-shell.c	2020-07-13 19:13:28.366724605 +0000
@@ -823,27 +823,8 @@
 ephy_embed_shell_setup_process_model (EphyEmbedShell *shell)
 {
   EphyEmbedShellPrivate *priv = ephy_embed_shell_get_instance_private (shell);
-  EphyPrefsProcessModel process_model;
-  guint max_processes;
-
-  if (ephy_embed_shell_get_mode (shell) == EPHY_EMBED_SHELL_MODE_APPLICATION)
-    process_model = EPHY_PREFS_PROCESS_MODEL_SHARED_SECONDARY_PROCESS;
-  else
-    process_model = g_settings_get_enum (EPHY_SETTINGS_MAIN, EPHY_PREFS_PROCESS_MODEL);
-
-  switch (process_model) {
-    case EPHY_PREFS_PROCESS_MODEL_SHARED_SECONDARY_PROCESS:
-      max_processes = 1;
-      break;
-    case EPHY_PREFS_PROCESS_MODEL_ONE_SECONDARY_PROCESS_PER_WEB_VIEW:
-      max_processes = g_settings_get_uint (EPHY_SETTINGS_MAIN, EPHY_PREFS_MAX_PROCESSES);
-      break;
-    default:
-      g_assert_not_reached ();
-  }
 
   webkit_web_context_set_process_model (priv->web_context, WEBKIT_PROCESS_MODEL_MULTIPLE_SECONDARY_PROCESSES);
-  webkit_web_context_set_web_process_count_limit (priv->web_context, max_processes);
 }
 
 static void
@@ -1040,11 +1025,6 @@
   webkit_security_manager_register_uri_scheme_as_secure (webkit_web_context_get_security_manager (priv->web_context),
                                                          "ephy-resource");
 
-  /* No support for FTP, try to open in nautilus instead of failing */
-  webkit_web_context_register_uri_scheme (priv->web_context, "ftp",
-                                          (WebKitURISchemeRequestCallback)ftp_request_cb,
-                                          NULL, NULL);
-
   /* Store cookies in moz-compatible SQLite format */
   cookie_manager = webkit_web_context_get_cookie_manager (priv->web_context);
   if (priv->mode != EPHY_EMBED_SHELL_MODE_INCOGNITO) {
