--- epiphany-3.29.1/lib/ephy-web-app-utils.c	2018-04-17 01:56:57.000000000 +0000
+++ epiphany-3.29.1/lib/ephy-web-app-utils.c	2018-09-19 13:59:48.028608513 +0000
@@ -203,7 +203,7 @@
                                  address);
   g_key_file_set_value (file, "Desktop Entry", "Exec", exec_string);
   g_free (exec_string);
-  g_key_file_set_value (file, "Desktop Entry", "StartupNotify", "true");
+  g_key_file_set_value (file, "Desktop Entry", "StartupNotify", "false");
   g_key_file_set_value (file, "Desktop Entry", "Terminal", "false");
   g_key_file_set_value (file, "Desktop Entry", "Type", "Application");
   g_key_file_set_value (file, "Desktop Entry", "Categories", "Network;GNOME;GTK;");
--- epiphany-3.29.1/data/org.gnome.Epiphany.desktop.in	2018-04-17 01:56:57.000000000 +0000
+++ epiphany-3.29.1/data/org.gnome.Epiphany.desktop.in	2018-09-19 13:59:37.393245202 +0000
@@ -5,7 +5,7 @@
 # Translators: Search terms to find this application. Do NOT translate or localize the semicolons! The list MUST also end with a semicolon!
 Keywords=web;browser;internet;
 Exec=epiphany %U
-StartupNotify=true
+StartupNotify=false
 Terminal=false
 Type=Application
 # Translators: Do NOT translate or transliterate this text (this is an icon file name)!
