--- a/embed/ephy-embed-prefs.c	2018-10-21 23:23:11.000000000 +0000
+++ b/embed/ephy-embed-prefs.c	2018-11-01 19:05:34.095588506 +0000
@@ -431,7 +431,7 @@
                                                        "enable-fullscreen", TRUE,
                                                        "enable-javascript", TRUE,
                                                        "enable-dns-prefetching", TRUE,
-                                                       "enable-mediasource", TRUE,
+                                                       "enable-mediasource", FALSE,
                                                        "javascript-can-open-windows-automatically", TRUE,
                                                        NULL);
 
