diff -Naur epiphany-42.0/src/bookmarks/ephy-bookmark-properties.c epiphany-41.3/src/bookmarks/ephy-bookmark-properties.c
--- epiphany-42.0/src/bookmarks/ephy-bookmark-properties.c	2022-03-17 19:42:44.000000000 +0000
+++ epiphany-41.3/src/bookmarks/ephy-bookmark-properties.c	2021-12-21 00:02:24.000000000 +0000
@@ -182,15 +182,13 @@
                                           GTK_ICON_SIZE_BUTTON);
     gtk_widget_set_margin_bottom (image, 8);
     gtk_widget_set_margin_top (image, 8);
-    gtk_box_pack_start (GTK_BOX (box), image, FALSE, TRUE, 0);
+    gtk_box_pack_start (GTK_BOX (box), image, FALSE, FALSE, 0);
   }
 
   label_text = default_tag ? EPHY_BOOKMARKS_FAVORITES_TAG : tag;
   label = gtk_label_new (label_text);
-  gtk_widget_set_hexpand (label, TRUE);
-  gtk_label_set_xalign (GTK_LABEL (label), 0);
   gtk_label_set_ellipsize (GTK_LABEL (label), PANGO_ELLIPSIZE_END);
-  gtk_box_pack_start (GTK_BOX (box), label, FALSE, TRUE, 0);
+  gtk_box_pack_start (GTK_BOX (box), label, FALSE, FALSE, 0);
 
   if (!default_tag) {
     GtkWidget *button;
@@ -200,8 +198,8 @@
                           gtk_image_new_from_icon_name ("window-close-symbolic",
                                                         GTK_ICON_SIZE_MENU));
     gtk_widget_set_valign (button, GTK_ALIGN_CENTER);
-    gtk_style_context_add_class (gtk_widget_get_style_context (button), "flat");
-    gtk_box_pack_start (GTK_BOX (box), button, FALSE, TRUE, 0);
+    gtk_button_set_relief (GTK_BUTTON (button), GTK_RELIEF_NONE);
+    gtk_box_pack_end (GTK_BOX (box), button, FALSE, FALSE, 0);
     g_signal_connect_object (button, "clicked",
                              G_CALLBACK (ephy_bookmark_properties_tag_widget_button_clicked_cb),
                              self,
diff -Naur epiphany-42.0/src/bookmarks/ephy-bookmarks-popover.c epiphany-41.3/src/bookmarks/ephy-bookmarks-popover.c
--- epiphany-42.0/src/bookmarks/ephy-bookmarks-popover.c	2022-03-17 19:42:44.000000000 +0000
+++ epiphany-41.3/src/bookmarks/ephy-bookmarks-popover.c	2021-12-21 00:02:24.000000000 +0000
@@ -235,6 +235,7 @@
   g_object_set (G_OBJECT (row), "height-request", 40, NULL);
 
   box = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 6);
+  gtk_widget_set_halign (box, GTK_ALIGN_START);
 
   if (g_strcmp0 (tag, EPHY_BOOKMARKS_FAVORITES_TAG) == 0) {
     image = gtk_image_new_from_icon_name ("emblem-favorite-symbolic", GTK_ICON_SIZE_MENU);
@@ -243,11 +244,10 @@
   }
   label = gtk_label_new (tag);
 
-  gtk_widget_set_hexpand (label, TRUE);
   gtk_label_set_ellipsize (GTK_LABEL (label), PANGO_ELLIPSIZE_END);
 
-  gtk_box_pack_start (GTK_BOX (box), image, FALSE, TRUE, 0);
-  gtk_box_pack_start (GTK_BOX (box), label, FALSE, TRUE, 0);
+  gtk_box_pack_start (GTK_BOX (box), image, FALSE, FALSE, 0);
+  gtk_box_pack_start (GTK_BOX (box), label, TRUE, FALSE, 0);
 
   gtk_container_add (GTK_CONTAINER (row), box);
   gtk_widget_show_all (row);
