diff --git a/meson.build b/meson.build
index d2e4ec8..f3fcc44 100644
--- a/meson.build
+++ b/meson.build
@@ -13,7 +13,6 @@ cc = meson.get_compiler('c')
 test_cflags = [
   '-funsigned-char',
   '-fstack-protector-strong',
-  '-fPIE',
   '-fPIC',
   '-Wconversion',
   '-Winline',
@@ -40,9 +39,6 @@ endif
 add_project_arguments(global_cflags, language: 'c')
 
 test_ldflags = [
-  '-pie',
-  '-Wl,-z,relro',
-  '-Wl,-z,now'
 ]
 global_ldflags = []
 foreach ldflag : test_ldflags
diff --git a/src/celluloid-main-window.c b/src/celluloid-main-window.c
index 86b9d90..8895cf1 100644
--- a/src/celluloid-main-window.c
+++ b/src/celluloid-main-window.c
@@ -717,12 +717,16 @@ celluloid_main_window_set_playlist_visible(	CelluloidMainWindow *wnd,
 		gint height;
 
 		gdk_window = gtk_widget_get_window(GTK_WIDGET(wnd));
+#if GTK_CHECK_VERSION(3,22,22)
 		window_state = gdk_window_get_state(gdk_window);
 		resize = window_state &
 			(	GDK_WINDOW_STATE_TOP_RESIZABLE |
 				GDK_WINDOW_STATE_RIGHT_RESIZABLE |
 				GDK_WINDOW_STATE_BOTTOM_RESIZABLE |
 				GDK_WINDOW_STATE_LEFT_RESIZABLE );
+#else
+		resize = TRUE;
+#endif
 		handle_pos =	gtk_paned_get_position
 				(GTK_PANED(priv->vid_area_paned));
 
diff --git a/src/meson.build b/src/meson.build
index b3d09ef..c321e4e 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -47,7 +47,7 @@ sources += custom_target('authors',
   ]
 )
 
-libgtk = dependency('gtk+-3.0', version: '>= 3.22.23')
+libgtk = dependency('gtk+-3.0', version: '>= 3.22.12')
 localedir = join_paths(get_option('prefix'), get_option('localedir'))
 cflags = [
   '-DG_SETTINGS_ENABLE_BACKEND',
