--- a/icu/source/configure.ac	2018-06-01 09:43:49.622228699 +0000
+++ b/icu/source/configure.ac	2018-06-01 09:53:52.918154206 +0000
@@ -1190,15 +1190,7 @@
     msg=none
 fi
 AC_MSG_RESULT($msg)
-AC_SUBST(ICULIBSUFFIX)
-if test "$ICULIBSUFFIX" != ""
-then
-    U_HAVE_LIB_SUFFIX=1
-    ICULIBSUFFIXCNAME=`echo _$ICULIBSUFFIX | sed 's/[^A-Za-z0-9_]/_/g'`
-    UCONFIG_CPPFLAGS="${UCONFIG_CPPFLAGS} -DU_HAVE_LIB_SUFFIX=1 -DU_LIB_SUFFIX_C_NAME=${ICULIBSUFFIXCNAME} "
-else
-    U_HAVE_LIB_SUFFIX=0
-fi
+U_HAVE_LIB_SUFFIX=0
 AC_SUBST(U_HAVE_LIB_SUFFIX)
 AC_SUBST(ICULIBSUFFIXCNAME)
 
--- a/icu/source/configure	2018-06-01 10:04:57.799038676 +0000
+++ a/icu/source/configure	2018-06-01 10:05:15.095759765 +0000
@@ -7599,14 +7599,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $msg" >&5
 $as_echo "$msg" >&6; }
 
-if test "$ICULIBSUFFIX" != ""
-then
-    U_HAVE_LIB_SUFFIX=1
-    ICULIBSUFFIXCNAME=`echo _$ICULIBSUFFIX | sed 's/^A-Za-z0-9_/_/g'`
-    UCONFIG_CPPFLAGS="${UCONFIG_CPPFLAGS} -DU_HAVE_LIB_SUFFIX=1 -DU_LIB_SUFFIX_C_NAME=${ICULIBSUFFIXCNAME} "
-else
-    U_HAVE_LIB_SUFFIX=0
-fi
+U_HAVE_LIB_SUFFIX=0
 
 
 
