Solaris doesn't have MAP_FILE

Not suitable for upstream

--- a/Source/WTF/wtf/FileSystem.cpp	2019-02-15 12:32:52.000000000 +0000
+++ b/Source/WTF/wtf/FileSystem.cpp	2019-02-15 12:51:12.458160543 +0000
@@ -316,7 +316,7 @@
         return;
     }
 
-    void* data = mmap(0, size, PROT_READ, MAP_FILE | MAP_SHARED, fd, 0);
+    void* data = mmap(0, size, PROT_READ, MAP_SHARED, fd, 0);
     close(fd);
 
     if (data == MAP_FAILED) {
