Solaris doesn't have MAP_FILE

Not suitable for upstream

--- a/Source/WTF/wtf/FileSystem.cpp	2019-08-12 11:01:21.125388685 +0000
+++ b/Source/WTF/wtf/FileSystem.cpp	2019-08-12 11:03:14.489142966 +0000
@@ -323,8 +323,12 @@
         return true;
     }
 
+#ifdef __sun__
+    void* data = mmap(0, size, PROT_READ, mode == MappedFileMode::Shared ? MAP_SHARED : MAP_PRIVATE, fd, 0);
+#else
     void* data = mmap(0, size, PROT_READ, MAP_FILE | (mode == MappedFileMode::Shared ? MAP_SHARED : MAP_PRIVATE), fd, 0);
-
+#endif
+    
     if (data == MAP_FAILED) {
         return false;
     }
