--- a/Source/WebCore/platform/glib/UserAgentGLib.cpp	2019-02-18 10:55:38.000000000 +0000
+++ b/Source/WebCore/platform/glib/UserAgentGLib.cpp	2019-08-11 16:01:16.500580373 +0000
@@ -83,12 +83,25 @@
 
     if (quirks.contains(UserAgentQuirks::NeedsMacintoshPlatform))
         uaString.append(UserAgentQuirks::stringForQuirk(UserAgentQuirks::NeedsMacintoshPlatform));
+    // Disabling Linux quirks for FreeBSD and Solaris; see below...
+#if !defined(__FreeBSD__) && !defined(__sun__)
     else if (quirks.contains(UserAgentQuirks::NeedsLinuxDesktopPlatform))
         uaString.append(UserAgentQuirks::stringForQuirk(UserAgentQuirks::NeedsLinuxDesktopPlatform));
+#endif
     else {
         uaString.append(platformForUAString());
         uaString.appendLiteral("; ");
         uaString.append(platformVersionForUAString());
+        // In addition to the native platform version, include Linux
+        // in the UA string, as FreeBSD does in its Chromium package.
+        // See https://bugs.webkit.org/show_bug.cgi?id=199745
+#if defined(__FreeBSD__) || defined(__sun__)
+#ifdef __sparc__
+        uaString.appendLiteral("; Linux sparc64");
+#else
+        uaString.appendLiteral("; Linux x86_64");
+#endif
+#endif
     }
 
     uaString.appendLiteral(") AppleWebKit/");
@@ -103,7 +116,7 @@
 
     // Version/X is mandatory *before* Safari/X to be a valid Safari UA. See
     // https://bugs.webkit.org/show_bug.cgi?id=133403 for details.
-    uaString.appendLiteral("Version/11.0 Safari/");
+    uaString.appendLiteral("Version/13.0 Safari/");
     uaString.append(versionForUAString());
 
     return uaString.toString();
--- a/Source/WebCore/platform/UserAgentQuirks.cpp	2019-04-08 11:24:42.000000000 +0000
+++ b/Source/WebCore/platform/UserAgentQuirks.cpp	2019-08-11 18:56:46.260689648 +0000
@@ -138,9 +138,9 @@
     switch (quirk) {
     case NeedsChromeBrowser:
         // Get versions from https://chromium.googlesource.com/chromium/src.git
-        return "Chrome/58.0.3029.81"_s;
+        return "Chrome/75.0.3770.141"_s;
     case NeedsMacintoshPlatform:
-        return "Macintosh; Intel Mac OS X 10_13_4"_s;
+        return "Macintosh; Intel Mac OS X 10_15"_s;
     case NeedsLinuxDesktopPlatform:
         return "X11; Linux x86_64"_s;
     case NumUserAgentQuirks:
