--- a/Source/WebCore/layout/integration/LayoutIntegrationLineIteratorLegacyPath.h	2020-10-10 16:48:21.000000000 +0000
+++ b/Source/WebCore/layout/integration/LayoutIntegrationLineIteratorLegacyPath.h	2020-10-10 19:28:05.599159682 +0000
@@ -28,6 +28,8 @@
 #include "LayoutIntegrationInlineContent.h"
 #include "RootInlineBox.h"
 
+#include "LayoutIntegrationRunIteratorLegacyPath.h"
+
 namespace WebCore {
 
 namespace LayoutIntegration {
--- a/Source/WebCore/layout/integration/LayoutIntegrationLineIterator.cpp	2020-10-10 19:27:24.607153242 +0000
+++ b/Source/WebCore/layout/integration/LayoutIntegrationLineIterator.cpp	2020-10-10 19:30:18.825359910 +0000
@@ -28,6 +28,8 @@
 
 #include "LayoutIntegrationLineLayout.h"
 
+#include "LayoutIntegrationRunIterator.h"
+
 namespace WebCore {
 namespace LayoutIntegration {
 
