glGetString can return null when GBM is disabled


--- a/Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp	2023-03-07 16:30:14.171175867 +0000
+++ b/Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp	2023-03-07 16:24:14.100031061 +0000
@@ -67,6 +67,9 @@
 #include <gst/gst.h>
 #endif
 
+#define makeString(s, ...) \
+    (__VA_OPT__(1 ||) s ? ::makeString(s __VA_OPT__(,) __VA_ARGS__) : "None"_s)
+
 namespace WebKit {
 using namespace WebCore;
 
