--- webkitgtk-2.42.0/Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreDMABuf.cpp-	2023-09-08 06:46:16.014341000 +0000
+++ webkitgtk-2.42.0/Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreDMABuf.cpp	2023-09-16 09:38:33.995315299 +0000
@@ -33,7 +33,6 @@
 #include "ShareableBitmap.h"
 #include "WebPageProxy.h"
 #include "WebProcessProxy.h"
-#include <WebCore/DMABufFormat.h>
 #include <WebCore/GLContext.h>
 #include <WebCore/IntRect.h>
 #include <WebCore/PlatformDisplay.h>
@@ -41,6 +40,7 @@
 #include <wtf/glib/GUniquePtr.h>
 
 #if USE(GBM)
+#include <WebCore/DMABufFormat.h>
 #include <gbm.h>
 #endif
 
@@ -120,7 +120,11 @@
             EGL_DMA_BUF_PLANE0_OFFSET_EXT, static_cast<EGLAttrib>(offset),
             EGL_DMA_BUF_PLANE0_PITCH_EXT, static_cast<EGLAttrib>(stride),
         };
-        if (modifier != uint64_t(WebCore::DMABufFormat::Modifier::Invalid) && display.eglExtensions().EXT_image_dma_buf_import_modifiers) {
+        if (
+#if USE(GBM)
+	    modifier != uint64_t(WebCore::DMABufFormat::Modifier::Invalid) &&
+#endif
+	    display.eglExtensions().EXT_image_dma_buf_import_modifiers) {
             std::array<EGLAttrib, 4> modifierAttributes {
                 EGL_DMA_BUF_PLANE0_MODIFIER_HI_EXT, static_cast<EGLAttrib>(modifier >> 32),
                 EGL_DMA_BUF_PLANE0_MODIFIER_LO_EXT, static_cast<EGLAttrib>(modifier & 0xffffffff),
--- webkitgtk-2.42.0/Source/WebKit/WebProcess/WebPage/gtk/AcceleratedSurfaceDMABuf.cpp-	2023-09-08 06:46:40.750383000 +0000
+++ webkitgtk-2.42.0/Source/WebKit/WebProcess/WebPage/gtk/AcceleratedSurfaceDMABuf.cpp	2023-09-15 22:34:15.065589417 +0000
@@ -31,13 +31,13 @@
 #include "ShareableBitmap.h"
 #include "WebPage.h"
 #include "WebProcess.h"
-#include <WebCore/DMABufFormat.h>
 #include <WebCore/PlatformDisplay.h>
 #include <array>
 #include <epoxy/egl.h>
 #include <wtf/SafeStrerror.h>
 
 #if USE(GBM)
+#include <WebCore/DMABufFormat.h>
 #include <WebCore/GBMDevice.h>
 #include <gbm.h>
 #endif
@@ -157,7 +157,11 @@
             EGL_DMA_BUF_PLANE0_OFFSET_EXT, static_cast<EGLAttrib>(metadata.offset),
             EGL_DMA_BUF_PLANE0_PITCH_EXT, static_cast<EGLAttrib>(metadata.stride),
         };
-        if (metadata.modifier != uint64_t(WebCore::DMABufFormat::Modifier::Invalid) && display.eglExtensions().EXT_image_dma_buf_import_modifiers) {
+        if (
+#if USE(GBM)
+	    metadata.modifier != uint64_t(WebCore::DMABufFormat::Modifier::Invalid) &&
+#endif
+	    display.eglExtensions().EXT_image_dma_buf_import_modifiers) {
             std::array<EGLAttrib, 4> modifierAttributes {
                 EGL_DMA_BUF_PLANE0_MODIFIER_HI_EXT, static_cast<EGLAttrib>(metadata.modifier >> 32),
                 EGL_DMA_BUF_PLANE0_MODIFIER_LO_EXT, static_cast<EGLAttrib>(metadata.modifier & 0xffffffff),
