--- a/Source/WebCore/platform/gtk/ScrollbarThemeGtk.cpp	2017-11-30 10:23:29.000000000 +0000
+++ b/Source/WebCore/platform/gtk/ScrollbarThemeGtk.cpp	2018-10-30 14:19:04.957446454 +0000
@@ -772,13 +772,18 @@
 
 ScrollbarButtonPressAction ScrollbarThemeGtk::handleMousePressEvent(Scrollbar&, const PlatformMouseEvent& event, ScrollbarPart pressedPart)
 {
+    gboolean warp_slider = FALSE;
     switch (pressedPart) {
     case BackTrackPart:
     case ForwardTrackPart:
-        if (event.button() == LeftButton)
-            return ScrollbarButtonPressAction::CenterOnThumb;
-        if (event.button() == RightButton)
-            return ScrollbarButtonPressAction::Scroll;
+        g_object_get(gtk_settings_get_default(),
+                     "gtk-primary-button-warps-slider",
+                     &warp_slider, NULL);
+        // shift key or middle/right button reverse sense of setting
+        return warp_slider ^ (event.shiftKey() ||
+                              event.button() != LeftButton)?
+                ScrollbarButtonPressAction::CenterOnThumb:
+                ScrollbarButtonPressAction::Scroll;
         break;
     case ThumbPart:
         if (event.button() != RightButton)
