--- a/meson_options.txt	2020-09-08 00:09:26.922830000 +0000
+++ b/meson_options.txt	2020-09-10 13:03:01.308156382 +0000
@@ -21,5 +21,6 @@
        description: 'Enable python formatters.')
 option('libpython-dir', type : 'string', value : '',
         description: 'Path to find libpythonXX.so')
+option('py-target', type : 'string', description : 'target Python')
 option('validate', type : 'feature', value : 'auto', yield: true,
        description: 'Enable GstValidate integration.')
--- a/meson.build	2025-08-16 12:56:17.840744580 +0000
+++ b/meson.build	2025-08-16 12:57:27.212900066 +0000
@@ -146,7 +146,7 @@
 
 pymod = import('python')
 python_opt = get_option('python')
-python = pymod.find_installation(required: python_opt)
+python = pymod.find_installation(get_option('py-target'), required: python_opt)
 has_python = false
 static_build = get_option('default_library') == 'static'
 if static_build
@@ -230,9 +230,6 @@
         endif
       endforeach
     endforeach
-    if not has_python
-      message('Could not find python library to load, will try loading at runtime')
-    endif
   endif
 endif
 
