--- a/build/make/Makefile	2018-01-24 22:25:44.000000000 +0000
+++ b/build/make/Makefile	2018-07-04 12:44:50.421173293 +0000
@@ -303,7 +303,7 @@
 	$(if $(quiet),@echo "    [LD] $$@")
 	$(qexec)$$(LD) -shared $$(LDFLAGS) \
             -Wl,--no-undefined -Wl,-soname,$$(SONAME) \
-            -Wl,--version-script,$$(EXPORTS_FILE) -o $$@ \
+            -Wl,-M,$$(EXPORTS_FILE) -o $$@ \
             $$(filter %.o,$$^) $$(extralibs)
 endef
 
