--- a/tests/check/Makefile.in	2019-04-19 00:09:15.000000000 +0000
+++ b/tests/check/Makefile.in	2019-04-29 16:09:36.959896640 +0000
@@ -3250,7 +3250,7 @@
 gst/gstelement.log: gst/gstelement$(EXEEXT)
 	@p='gst/gstelement$(EXEEXT)'; \
 	b='gst/gstelement'; \
-	$(am__check_pre) $(LOG_DRIVER) --test-name "$$f" \
+	$(am__check_pre) CK_FORK=no $(LOG_DRIVER) --test-name "$$f" \
 	--log-file $$b.log --trs-file $$b.trs \
 	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
 	"$$tst" $(AM_TESTS_FD_REDIRECT)
