--- a/plugins/elements/gstelements_private.c	2018-11-28 09:45:49.000000000 +0000
+++ b/plugins/elements/gstelements_private.c	2019-09-05 19:25:26.129202441 +0000
@@ -133,7 +133,11 @@
   gssize written;
 
 #ifdef HAVE_SYS_UIO_H
+#ifdef __sun__
+  if (iovcnt <= UIOA_IOV_MAX) {
+#else
   if (iovcnt <= UIO_MAXIOV) {
+#endif
     do {
       written = writev (fd, iov, iovcnt);
     } while (written < 0 && errno == EINTR);
