--- a/Source/WebCore/platform/mediastream/MediaStreamTrackPrivate.h	2024-09-28 19:04:53.803251036 +0000
+++ b/Source/WebCore/platform/mediastream/MediaStreamTrackPrivate.h	2024-09-28 19:05:52.989603879 +0000
@@ -149,8 +149,8 @@
 
     void setIdForTesting(String&& id) { m_id = WTFMove(id); }
 
+    const Logger& logger() const { return m_logger; }
 #if !RELEASE_LOG_DISABLED
-    const Logger& logger() const final { return m_logger; }
     uint64_t logIdentifier() const final { return m_logIdentifier; }
 #endif
 
