--- webkit/Source/WTF/wtf/glib/FileSystemGlib.cpp.~2~	2024-04-07 23:02:21.909379687 +0000
+++ webkit/Source/WTF/wtf/glib/FileSystemGlib.cpp	2024-04-08 10:38:37.794531247 +0000
@@ -74,6 +74,18 @@
 {
     return { };
 }
+#elif OS(SOLARIS)
+CString currentExecutablePath()
+{
+    static char readLinkBuffer[PATH_MAX];
+    static char path[PATH_MAX];
+    sprintf(path, "/proc/%ld/path/a.out", (long)getpid());
+    ssize_t result = readlink(path, readLinkBuffer, PATH_MAX);
+    if (result == -1)
+        return { };
+    readLinkBuffer[result] = '\0';
+    return CString(readLinkBuffer);
+}
 #elif OS(UNIX)
 CString currentExecutablePath()
 {
