--- webkit/Source/JavaScriptCore/runtime/JSCBytecodeCacheVersion.cpp~	2024-10-06 16:47:40.000000000 +0000
+++ webkit/Source/JavaScriptCore/runtime/JSCBytecodeCacheVersion.cpp	2024-10-06 18:00:53.317868029 +0000
@@ -66,7 +66,7 @@
         }
         cacheVersion.construct(0);
         dataLogLnIf(JSCBytecodeCacheVersionInternal::verbose, "Failed to get UUID for JavaScriptCore framework");
-#elif OS(UNIX) && !PLATFORM(PLAYSTATION)
+#elif OS(UNIX) && !PLATFORM(PLAYSTATION) && !OS(SOLARIS)
         auto result = ([&] -> std::optional<uint32_t> {
             Dl_info info { };
             if (!dladdr(jsFunctionAddr, &info))
