--- webkitgtk-2.52.0/Source/JavaScriptCore/runtime/JSCBytecodeCacheVersion.cpp.~1~	2026-03-20 12:28:00.040331167 +0000
+++ webkitgtk-2.52.0/Source/JavaScriptCore/runtime/JSCBytecodeCacheVersion.cpp	2026-03-20 12:28:45.166313205 +0000
@@ -68,7 +68,7 @@
         }
         cacheVersion.construct(0);
         dataLogLnIf(JSCBytecodeCacheVersionInternal::verbose, "Failed to get UUID for JavaScriptCore framework");
-#elif OS(UNIX) && !PLATFORM(PLAYSTATION) && !OS(HAIKU) && !OS(QNX)
+#elif OS(UNIX) && !PLATFORM(PLAYSTATION) && !OS(HAIKU) && !OS(QNX) && !OS(SOLARIS)
         auto result = ([&] -> std::optional<uint32_t> {
             Dl_info info { };
             if (!dladdr(jsFunctionAddr, &info))
