--- webkit/Source/WebCore/page/Page.cpp~	2025-02-18 16:11:02.000000000 +0000
+++ webkit/Source/WebCore/page/Page.cpp	2025-02-18 19:12:53.418659201 +0000
@@ -539,8 +539,10 @@
     if (RefPtr scrollingCoordinator = m_scrollingCoordinator)
         scrollingCoordinator->pageDestroyed();
 
+#if ENABLE(RESOURCE_USAGE)
     if (RefPtr resourceUsageOverlay = m_resourceUsageOverlay)
         resourceUsageOverlay->detachFromPage();
+#endif
 
     checkedBackForward()->close();
     if (!isUtilityPage())
