--- webkit/Source/JavaScriptCore/CMakeLists.txt.~1~	2024-08-10 18:10:46.000000000 +0000
+++ webkit/Source/JavaScriptCore/CMakeLists.txt	2024-08-11 19:29:59.435962676 +0000
@@ -1446,7 +1446,7 @@
 if (COMPILER_IS_GCC_OR_CLANG)
     # Avoid using fused multiply-add instructions since this could give different results
     # for e.g. parseInt depending on the platform and compilation flags.
-    if (WTF_CPU_LOONGARCH64)
+    if (WTF_CPU_LOONGARCH64 OR (COMPILER_IS_CLANG AND "${CMAKE_SYSTEM_NAME}" STREQUAL "SunOS"))
         WEBKIT_ADD_TARGET_CXX_FLAGS(JavaScriptCore -ffp-contract=off)
     else ()
         WEBKIT_ADD_TARGET_CXX_FLAGS(JavaScriptCore -ffp-contract=off -fno-slp-vectorize)
