--- webkit/Source/WTF/wtf/posix/FileSystemPOSIX.cpp.~1~	2025-03-27 15:54:54.289855895 +0000
+++ webkit/Source/WTF/wtf/posix/FileSystemPOSIX.cpp	2025-03-27 16:00:54.305198539 +0000
@@ -178,8 +178,13 @@
     auto buffer = MallocSpan<char>::malloc(length);
     snprintf(buffer.mutableSpan().data(), length, "%s/%s-XXXXXX", directory, prefixUTF8.data());
 
+#if defined(__sun)
+    auto handle = FileHandle::adopt(mkstemp(buffer.mutableSpan().data()));
+    if (!handle || fcntl(handle.platformHandle(), F_SETFD, FD_CLOEXEC) < 0)
+#else
     auto handle = FileHandle::adopt(mkostemp(buffer.mutableSpan().data(), O_CLOEXEC));
     if (!handle)
+#endif
         return { String(), FileHandle() };
 
     return { String::fromUTF8(buffer.span().data()), WTFMove(handle) };
