--- webkitgtk-2.46.1/Source/cmake/OptionsGTK.cmake.~1~	2024-09-30 18:51:59.030494706 +0000
+++ webkitgtk-2.46.1/Source/cmake/OptionsGTK.cmake	2024-09-30 18:52:49.517613690 +0000
@@ -67,7 +67,7 @@
 
 # Private options specific to the GTK port. Changing these options is
 # completely unsupported. They are intended for use only by WebKit developers.
-WEBKIT_OPTION_DEFINE(USE_SYSPROF_CAPTURE "Whether to use libsysprof-capture for tracing." PRIVATE ON)
+WEBKIT_OPTION_DEFINE(USE_SYSPROF_CAPTURE "Whether to use libsysprof-capture for tracing." PRIVATE OFF)
 
 WEBKIT_OPTION_DEPEND(USE_SYSTEM_SYSPROF_CAPTURE USE_SYSPROF_CAPTURE)
 
