--- webkit/Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp.~1~	2023-09-16 12:34:13.000000000 +0000
+++ webkit/Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp	2024-05-19 15:35:30.859319612 +0000
@@ -89,6 +89,7 @@
     return IPC::PlatformConnectionOptions::SetCloexecOnClient | IPC::PlatformConnectionOptions::SetCloexecOnServer;
 }
 
+#if OS(LINUX)
 static bool isSandboxEnabled(const ProcessLauncher::LaunchOptions& launchOptions)
 {
 #if !USE(SYSTEM_MALLOC)
@@ -116,6 +117,7 @@
 
     return launchOptions.extraInitializationData.get<HashTranslatorASCIILiteral>("enable-sandbox"_s) == "true"_s;
 }
+#endif
 
 void ProcessLauncher::launchProcess()
 {
