--- webkitgtk-2.52.0/Source/WebCore/platform/graphics/gstreamer/VideoFrameGStreamer.cpp~	2026-03-18 09:41:39.519486700 +0000
+++ webkitgtk-2.52.0/Source/WebCore/platform/graphics/gstreamer/VideoFrameGStreamer.cpp	2026-03-20 15:28:02.544540511 +0000
@@ -599,7 +599,9 @@
         inputSample = self.sample();
         break;
     case VideoFrameGStreamer::MemoryType::GL:
+#if USE(GBM)
     case VideoFrameGStreamer::MemoryType::DMABuf:
+#endif
         inputSample = self.downloadSample();
         break;
     }
