--- webkit/Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp.~2~	2024-08-11 08:45:54.597125650 +0000
+++ webkit/Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp	2024-08-11 10:55:44.301101581 +0000
@@ -210,7 +210,9 @@
     argv[i++] = const_cast<char*>(realExecutablePath.data());
     argv[i++] = processIdentifier.get();
     argv[i++] = webkitSocket.get();
+#if OS(LINUX)
     argv[i++] = pidSocket.get();
+#endif
 #if ENABLE(DEVELOPER_MODE)
     if (configureJSCForTesting)
         argv[i++] = const_cast<char*>("--configure-jsc-for-testing");
