--- openvpn-2.5.6/doc/Makefile.am~	2022-03-16 09:03:20.000000000 +0000
+++ openvpn-2.5.6/doc/Makefile.am	2022-09-05 17:55:04.885163824 +0000
@@ -74,6 +74,7 @@
 	$(RST2MAN) $< > $@
 else
 	@echo "Missing python-docutils - skipping man page generation ($@)"
+	touch $@
 endif
 
 .8.rst.8.html .5.rst.5.html :
--- openvpn-2.5.6/doc/openvpn.8.rst.~2~	2022-09-05 17:46:07.219303209 +0000
+++ openvpn-2.5.6/doc/openvpn.8.rst	2022-09-05 17:57:10.108802389 +0000
@@ -133,7 +133,6 @@
 SEE ALSO
 ========
 
-``openvpn-examples``\(5),
 ``in.dhcpcd``\(1m),
 ``ifconfig``\(1m),
 ``openssl``\(1openssl),
