--- a/configure	2016-02-13 21:05:42.000000000 +0000
+++ b/configure	2018-07-10 16:33:35.460225176 +0000
@@ -41,6 +41,8 @@
 # Store the configure line that was used
 configuration="$*"
 
+ROOT_DIR=$(dirname "$0")
+
 # utility functions
 tolower() {
   tr '[A-Z]' '[a-z]'
@@ -1532,7 +1534,7 @@
 fi
 
 list_subparts() {
-  test ! -e ffmpeg/libav${3} && return 1
+  test ! -e $ROOT_DIR/ffmpeg/libav${3} && return 1
   pattern="s/^[^#]*${1}.*([^ ,]*, *\([^ ,)]*\).*/\1_${2}/p"
   return 0
 }
@@ -1922,7 +1924,7 @@
 elif x86; then
   # all other OSes try to extract CPU information from a small helper
   # program cpuinfo instead
-  $_host_cc -o cpuinfo$_exesuf cpuinfo.c
+  $_host_cc -o cpuinfo$_exesuf $ROOT_DIR/cpuinfo.c
   _cpuinfo="./cpuinfo$_exesuf"
 fi
 
@@ -8203,7 +8205,7 @@
   echores "docbook.xsl"
 fi
 
-cat > DOCS/xml/html-chunk.xsl << EOF
+cat > $ROOT_DIR/DOCS/xml/html-chunk.xsl << EOF
 <?xml version="1.0" encoding="utf-8"?>
 <!-- **************************************************
      This file is generated automatically. DO NOT EDIT.
@@ -8217,7 +8219,7 @@
 </xsl:stylesheet>
 EOF
 
-cat > DOCS/xml/html-single.xsl << EOF
+cat > $ROOT_DIR/DOCS/xml/html-single.xsl << EOF
 <?xml version="1.0" encoding="utf-8"?>
 <!-- **************************************************
      This file is generated automatically. DO NOT EDIT.
@@ -8256,7 +8258,7 @@
 fi
 
 for lang in $language_doc; do
-  cat > DOCS/xml/$lang/main.xml << EOF
+  cat > $ROOT_DIR/DOCS/xml/$lang/main.xml << EOF
 <?xml version="1.0" encoding="utf-8" standalone="no"?>
 <!-- **************************************************
      This file is generated automatically. DO NOT EDIT.
