--- pitivi-0.999.0/meson.build	2018-08-29 14:33:20.000000000 +0000
+++ pitivi-0.999.0/meson.build	2020-08-24 20:28:18.471265211 +0000
@@ -23,9 +23,6 @@
 cairo_dep = dependency('cairo')
 pycairo_dep = dependency('py3cairo')
 
-gst_transcoder_dep = dependency('gst-transcoder-1.0', version : '>= 1.8.1',
-    fallback : ['gst-transcoder', 'gst_transcoder_dep'])
-
 pkgdatadir = join_paths(get_option('datadir'), meson.project_name())
 
 git = find_program('git', required : false)
